hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common/src/main/java/org/apache/hadoop/yarn/server/federation/utils/FederationStateStoreFacade.java

```diff
@@ -195,11 +194,23 @@ public static RetryPolicy createRetryPolicy(Configuration conf) {
 
   /**
    * Returns the singleton instance of the FederationStateStoreFacade object.
+   * @param conf Configuration.
    *
    * @return the singleton {@link FederationStateStoreFacade} instance
    */
-  public static FederationStateStoreFacade getInstance() {
-    return FACADE;
+  public static FederationStateStoreFacade getInstance(Configuration conf) {
+    if(facade == null) {
```

Comment:
Can we exit early?
```
if (facade != null) {
  return facade;
}
```

Reply:
Fix spaces too.

Reply:
Thank you very much for your help in reviewing the code! I will improve this part of the code.